smart-contracts/contracts/mixins/MixinKeys.sol

```diff
@@ -154,12 +155,35 @@ contract MixinKeys is MixinErrors, MixinLockCore {
   /**
    * Migrate data from the previous single owner => key mapping to
    * the new data structure w multiple tokens.
-   * No data migration needed for v10 > v11
    */
-  function migrate(bytes calldata) public virtual {
-    schemaVersion = publicLockVersion();
-  }
+  function migrate(bytes calldata _calldata) public virtual {
+    // make sure we have correct data version before migrating
+    require(
+      (
+        (schemaVersion == publicLockVersion() - 1)
+        ||
+        schemaVersion == 0
+      ),
+      'SCHEMA_VERSION_NOT_CORRECT'
+    );
+
+    // only for mainnet
+    if(block.chainid == 1) {
 
+      // TODO: hardcode mainnet Unlock address
+      address newUnlockAddress = 0x7039d2BB4CfC5f5DA49E6b4b9c40400bccb0d1E8;
```

Comment:
How will we actually handle this?

Reply:
We need to first deploy the new version of Unlock on mainnet (and configure it with a `configUnlock`) then we add hardcode the address here.

Reply:
Ok, that's helpful. Since we want to add more changes to the `Unlock` contract (fees...) we can only deploy after that, correct?